gui/packages/ubuntupro/lib/pages/subscribe_now/subscribe_now_model.dart

```diff
@@ -35,8 +53,25 @@ class SubscribeNowModel {
     }
   }
 
-  /// Returns true if the environment variable 'UP4W_ALLOW_STORE_PURCHASE' has been set.
-  /// Since this reading won't change during the app lifetime, even if the user changes
-  /// it's value from outside, the value is cached so we don't check the environment more than once.
-  bool get purchaseAllowed => _isPurchaseAllowed;
+  void tokenUpdate(String token) async {
+    _token.update(token);
+    notifyListeners();
+  }
+}
+
+/// A value-notifier for the [ProToken] with validation.
+class ProTokenValue extends EitherValueNotifier<TokenError, ProToken?> {
```

Comment:
I suspect we don't need to have this as a ValueNotifier if the SubscribeNowModel is a ChangeNotifier.